kafka/producer/base.py

```diff
@@ -111,12 +160,16 @@ def __init__(self, client, async=False,
                  codec=None,
                  batch_send=False,
                  batch_send_every_n=BATCH_SEND_MSG_COUNT,
-                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL):
+                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL,
+                 async_retry_options=ASYNC_RETRY_OPTIONS,
```

Comment:
i agree that lots of kwargs gets messy, but a typical user will generally only want/need to set a small set of the all available kwargs. One way to make large kwargs lists easier to maintain is to keep a module or class level dictionary of defaults, and just put `__init__(client, **kwargs)`. Then in the constructor you could loop through the dictionary of defaults, checking for an explicit init value in `kwargs` . This is how the KafkaConsumer constructor manages kwargs options.